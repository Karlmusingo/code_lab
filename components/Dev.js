import React, { Component } from "react";
import {
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
  StyleSheet
} from "react-native";
import { Right } from "native-base";

class Dev extends Component {
  state = {};

  goToProfile = dev => {
    this.props.navigation.navigate("Profile", { dev });
  };

  render() {
    const { dev } = this.props;

    return (
      <TouchableWithoutFeedback onPress={() => this.goToProfile(dev)}>
        <View style={[styles.dev]}>
          <Image source={{ uri: dev.avatarUrl }} style={styles.image} />
          <View style={styles.names}>
            <Text style={{ fontSize: 20, fontWeight: "400" }}>{dev.name}</Text>
            <Text style={{ marginTop: 8 }}>@{dev.login}</Text>
          </View>
          <View style={styles.badge}>
            <Text style={styles.textBadge}>{dev.repositories.totalCount}</Text>
          </View>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = StyleSheet.create({
  dev: {
    backgroundColor: "#F0F0F0",
    borderBottomColor: "#8492A6",
    borderBottomWidth: 1,
    paddingRight: 15,
    paddingLeft: 15,
    width: "100%",
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  image: {
    width: 70,
    height: 70,
    margin: 8,
    borderRadius: 35
  },
  names: {
    width: 190,
    display: "flex",
    justifyContent: "center"
  },
  badge: {
    backgroundColor: "#15cd66",
    height: 25,
    borderRadius: 13,
    paddingRight: 3
  },
  textBadge: {
    color: "#fff",
    padding: 5
  }
});

export default Dev;
