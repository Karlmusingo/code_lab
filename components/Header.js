import React, { Component } from "react";
import { Query } from "react-apollo";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import {
  View,
  Text,
  Image,
  TextInput,
  StyleSheet,
  Platform,
  FlatList,
  AsyncStorage,
  TouchableWithoutFeedback,
  TouchableHighlight,
  StatusBar,
  NativeModules
} from "react-native";
import { Icon } from "react-native-elements";
import ElevatedView from "react-native-elevated-view";

const { StatusBarManager } = NativeModules;
const STATUSBAR_HEIGHT = Platform.OS === "ios" ? 20 : StatusBarManager.HEIGHT;

export class Header extends Component {
  state = {
    displaySearch: false,
    displayDropdown: false,
    dropdownMenu: [{ key: "Logout" }]
  };

  displaySearch = () => {
    const { displaySearch } = this.state;
    this.props.search("");
    if (displaySearch) this.textInput.clear();
    this.setState({ displaySearch: !displaySearch });
  };

  displayDropdown = () => {
    this.setState({ displayDropdown: !this.state.displayDropdown });
  };

  logout = async () => {
    this.displayDropdown();
    this.props.navigation.navigate("Home");
    await AsyncStorage.clear();
  };

  render() {
    const { isLoggedIn, data, search } = this.props;

    return (
      <View style={styles.header}>
        <StatusBar />
        <TouchableWithoutFeedback onPress={this.displayDropdown}>
          {isLoggedIn ? (
            <View
              style={[styles.avatar, { display: isLoggedIn ? "flex" : "none" }]}
            >
              <Image
                source={
                  data.viewer
                    ? { uri: data.viewer.avatarUrl }
                    : { uri: "image" }
                }
                style={styles.image}
                onPress={this.displayDropdown}
              />
              <Icon name="sort-down" color="#000" type="font-awesome" />
            </View>
          ) : (
            <Text />
          )}
        </TouchableWithoutFeedback>
        <ElevatedView
          elevation={10}
          style={[
            styles.dropdown
            // { display: this.state.displayDropdown ? "flex" : "none" }
          ]}
        >
          <FlatList
            style={{ display: this.state.displayDropdown ? "flex" : "none" }}
            ItemSeparatorComponent={({ highlighted }) => (
              <View
                style={[styles.separator, highlighted && { marginLeft: 0 }]}
              />
            )}
            data={[...this.state.dropdownMenu]}
            renderItem={({ item }) => (
              <TouchableHighlight
                onPress={item => {
                  this.logout();
                }}
              >
                <View
                  style={{
                    backgroundColor: "#fff",
                    width: 100
                  }}
                >
                  <Text style={styles.dropdownItem}>{item.key}</Text>
                </View>
              </TouchableHighlight>
            )}
          />
        </ElevatedView>
        <Text style={styles.banner}>Code Lab</Text>
        <View style={styles.search}>
          <TextInput
            style={[
              styles.searchInput,
              { display: this.state.displaySearch ? "flex" : "none" }
            ]}
            ref={input => {
              this.textInput = input;
            }}
            placeholder="Search..."
            onChangeText={text => search(text)}
          />
        </View>
        {isLoggedIn ? (
          <Icon
            style={styles.searchIcon}
            name="search"
            onPress={this.displaySearch}
            color="#929292"
            type="font-awesome"
          />
        ) : (
          <Text />
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  header: {
    borderBottomColor: "#929292",
    borderBottomWidth: 1,
    paddingHorizontal: 17,
    paddingTop: Platform.OS === "android" ? STATUSBAR_HEIGHT : 0,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    zIndex: 9
  },
  banner: {
    fontWeight: "700",
    fontSize: 25
  },
  avatar: {
    width: 70,
    height: 60,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between"
  },
  image: {
    width: 50,
    height: 50,
    marginRight: 5,
    borderRadius: 25,
    left: 0,
    top: 0
  },
  search: {
    display: "flex",
    flexDirection: "row",
    position: "absolute",
    right: 14,
    top: Platform.OS === "android" ? STATUSBAR_HEIGHT + 14 : 14
  },
  searchInput: {
    height: 35,
    width: 250,
    borderColor: "gray",
    backgroundColor: "#F0F0F0",
    borderWidth: 1,
    borderRadius: 7,
    padding: 5,
    display: "none",
    right: 0,
    top: 0
  },
  searchIcon: {
    zIndex: 999,
    marginLeft: 20,
    marginRight: 70
  },
  dropdown: {
    position: "absolute",
    display: "flex",
    top: Platform.OS === "android" ? STATUSBAR_HEIGHT + 62 : 62,
    left: 15,
    paddingLeft: 5
  },
  dropdownItem: {
    fontSize: 20,
    padding: 5,
    paddingLeft: 5
  },
  separator: {
    backgroundColor: "#fff",
    borderBottomColor: "#8492A6",
    borderBottomWidth: 1,
    padding: 3,
    width: 100
  }
});

export const VIEWER_QUERY = gql`
  query {
    viewer {
      avatarUrl
    }
  }
`;

export default graphql(VIEWER_QUERY)(Header);
