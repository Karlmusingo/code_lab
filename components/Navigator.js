import { createAppContainer, createStackNavigator } from "react-navigation";
import HomeScreen from "../screens/Home";
// import ListScreen from "../screens/ListDevs";

export default createAppContainer(
  createStackNavigator(
    {
      Home: HomeScreen,
      List: ListScreen
    },
    {
      initialRouteName: "Home"
    }
  )
);
