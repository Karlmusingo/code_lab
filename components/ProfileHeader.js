import React, { Component } from "react";
import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight,
  Dimensions,
  Share
} from "react-native";
import { Icon } from "react-native-elements";

class ProfileHeader extends Component {
  state = {};
  onShare = async () => {
    try {
      const { login } = this.props;
      const result = await Share.share({
        message: `https://github.com/${login}`,
        title: "Share profile"
      });

      if (result.action === Share.sharedAction) {
        if (result.activityType) {
          // shared with activity type of result.activityType
        } else {
          // shared
        }
      } else if (result.action === Share.dismissedAction) {
        // dismissed
      }
    } catch (error) {
      alert(error.message);
    }
  };

  render() {
    const { goBack } = this.props;
    return (
      <View style={styles.header}>
        <View>
          <TouchableHighlight onPress={goBack}>
            <Icon
              // style={styles.searchIcon}
              name="chevron-left"
              onPress={this.displaySearch}
              color="#fff"
              type="font-awesome"
            />
          </TouchableHighlight>
        </View>
        <View>
          <Text style={styles.title}>Code Lab</Text>
        </View>
        <View>
          <TouchableHighlight>
            <Icon
              // style={styles.searchIcon}
              name="share-alt"
              onPress={this.onShare}
              color="#fff"
              type="font-awesome"
              size={30}
            />
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  header: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    paddingHorizontal: 17,
    position: "absolute",
    top: 40,
    width: "100%",
    zIndex: 9
    // backgroundColor: "#fff"
  },
  title: {
    fontSize: 22,
    fontWeight: "700",
    color: "#fff"
  }
});

export default ProfileHeader;
