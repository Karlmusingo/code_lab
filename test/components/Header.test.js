import React from "react";
import { shallow, mount } from "enzyme";
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";
import { Header, VIEWER_QUERY } from "../../components/Header";

const mocks = [
  {
    request: {
      query: VIEWER_QUERY
    },
    result: {
      data: {
        viewer: { avatarUrl: "http://avatarUrl.jpg" }
      }
    }
  }
];

describe("Header", () => {
  describe("Rendering", () => {
    const props = {
      data: {
        viewer: {
          avatarUrl: "avatarurl"
        }
      },
      isLoggedIn: true,
      search: jest.fn()
    };
    xit("should match to snapshot", () => {
      const component = renderer
        .create(
          <MockedProvider mocks={mocks} addTypename={false}>
            <Header />
          </MockedProvider>
        )
        .toJSON();

      expect(component).toMatchSnapshot();
    });

    xit("should call the search prop", () => {
      const search = jest.fn();
      const component = renderer.create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Header isLoggedIn={true} search={search} />
        </MockedProvider>
      );

      const textInput = component.root.findByType("TextInput");
      textInput.props.onChangeText("text");

      expect(search).toHaveBeenCalledWith("text");
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should display the search bar", () => {
      const component = shallow(<Header {...props} />);
      const instance = component.instance();
      instance.displaySearch();
      expect(component).toMatchSnapshot();
    });

    it("should display the search bar", () => {
      const component = shallow(<Header {...props} />);
      const instance = component.instance();
      instance.displayDropdown();
      expect(component).toMatchSnapshot();
    });

    it("should display the search bar", () => {
      const component = shallow(<Header {...props} />);
      const instance = component.instance();
      instance.logout();
      expect(component).toMatchSnapshot();
    });

    it("should type in search bar", () => {
      const component = shallow(<Header {...props} />);
      const input = component.find("TextInput").simulate("changeText", "text");
      expect(component).toMatchSnapshot();
    });
  });
});
