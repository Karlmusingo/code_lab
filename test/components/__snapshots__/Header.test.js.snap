// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering should display the search bar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#929292",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 17,
      "paddingTop": 0,
      "zIndex": 9,
    }
  }
>
  <StatusBar
    animated={false}
    showHideTransition="fade"
  />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "width": 70,
          },
          Object {
            "display": "flex",
          },
        ]
      }
    >
      <Image
        onPress={[Function]}
        source={
          Object {
            "uri": "avatarurl",
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "left": 0,
            "marginRight": 5,
            "top": 0,
            "width": 50,
          }
        }
      />
      <Themed.Icon
        color="#000"
        name="sort-down"
        type="font-awesome"
      />
    </View>
  </TouchableWithoutFeedback>
  <ElevatedView
    elevation={10}
    style={
      Array [
        Object {
          "display": "flex",
          "left": 15,
          "paddingLeft": 5,
          "position": "absolute",
          "top": 62,
        },
      ]
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "key": "Logout",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "display": "none",
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ElevatedView>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
      }
    }
  >
    Code Lab
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "position": "absolute",
        "right": 14,
        "top": 14,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search..."
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#F0F0F0",
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "display": "none",
            "height": 35,
            "padding": 5,
            "right": 0,
            "top": 0,
            "width": 250,
          },
          Object {
            "display": "flex",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Themed.Icon
    color="#929292"
    name="search"
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 20,
        "marginRight": 70,
        "zIndex": 999,
      }
    }
    type="font-awesome"
  />
</View>
`;

exports[`Header Rendering should display the search bar 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#929292",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 17,
      "paddingTop": 0,
      "zIndex": 9,
    }
  }
>
  <StatusBar
    animated={false}
    showHideTransition="fade"
  />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "width": 70,
          },
          Object {
            "display": "flex",
          },
        ]
      }
    >
      <Image
        onPress={[Function]}
        source={
          Object {
            "uri": "avatarurl",
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "left": 0,
            "marginRight": 5,
            "top": 0,
            "width": 50,
          }
        }
      />
      <Themed.Icon
        color="#000"
        name="sort-down"
        type="font-awesome"
      />
    </View>
  </TouchableWithoutFeedback>
  <ElevatedView
    elevation={10}
    style={
      Array [
        Object {
          "display": "flex",
          "left": 15,
          "paddingLeft": 5,
          "position": "absolute",
          "top": 62,
        },
      ]
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "key": "Logout",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "display": "flex",
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ElevatedView>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
      }
    }
  >
    Code Lab
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "position": "absolute",
        "right": 14,
        "top": 14,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search..."
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#F0F0F0",
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "display": "none",
            "height": 35,
            "padding": 5,
            "right": 0,
            "top": 0,
            "width": 250,
          },
          Object {
            "display": "none",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Themed.Icon
    color="#929292"
    name="search"
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 20,
        "marginRight": 70,
        "zIndex": 999,
      }
    }
    type="font-awesome"
  />
</View>
`;

exports[`Header Rendering should display the search bar 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#929292",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 17,
      "paddingTop": 0,
      "zIndex": 9,
    }
  }
>
  <StatusBar
    animated={false}
    showHideTransition="fade"
  />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "width": 70,
          },
          Object {
            "display": "flex",
          },
        ]
      }
    >
      <Image
        onPress={[Function]}
        source={
          Object {
            "uri": "avatarurl",
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "left": 0,
            "marginRight": 5,
            "top": 0,
            "width": 50,
          }
        }
      />
      <Themed.Icon
        color="#000"
        name="sort-down"
        type="font-awesome"
      />
    </View>
  </TouchableWithoutFeedback>
  <ElevatedView
    elevation={10}
    style={
      Array [
        Object {
          "display": "flex",
          "left": 15,
          "paddingLeft": 5,
          "position": "absolute",
          "top": 62,
        },
      ]
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "key": "Logout",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "display": "flex",
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ElevatedView>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
      }
    }
  >
    Code Lab
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "position": "absolute",
        "right": 14,
        "top": 14,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search..."
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#F0F0F0",
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "display": "none",
            "height": 35,
            "padding": 5,
            "right": 0,
            "top": 0,
            "width": 250,
          },
          Object {
            "display": "none",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Themed.Icon
    color="#929292"
    name="search"
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 20,
        "marginRight": 70,
        "zIndex": 999,
      }
    }
    type="font-awesome"
  />
</View>
`;

exports[`Header Rendering should type in search bar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#929292",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 17,
      "paddingTop": 0,
      "zIndex": 9,
    }
  }
>
  <StatusBar
    animated={false}
    showHideTransition="fade"
  />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "width": 70,
          },
          Object {
            "display": "flex",
          },
        ]
      }
    >
      <Image
        onPress={[Function]}
        source={
          Object {
            "uri": "avatarurl",
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "left": 0,
            "marginRight": 5,
            "top": 0,
            "width": 50,
          }
        }
      />
      <Themed.Icon
        color="#000"
        name="sort-down"
        type="font-awesome"
      />
    </View>
  </TouchableWithoutFeedback>
  <ElevatedView
    elevation={10}
    style={
      Array [
        Object {
          "display": "flex",
          "left": 15,
          "paddingLeft": 5,
          "position": "absolute",
          "top": 62,
        },
      ]
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "key": "Logout",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "display": "none",
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ElevatedView>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "700",
      }
    }
  >
    Code Lab
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "position": "absolute",
        "right": 14,
        "top": 14,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search..."
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#F0F0F0",
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "display": "none",
            "height": 35,
            "padding": 5,
            "right": 0,
            "top": 0,
            "width": 250,
          },
          Object {
            "display": "none",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Themed.Icon
    color="#929292"
    name="search"
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 20,
        "marginRight": 70,
        "zIndex": 999,
      }
    }
    type="font-awesome"
  />
</View>
`;
